mllib/src/test/scala/org/apache/spark/ml/recommendation/ALSSuite.scala

```diff
@@ -512,6 +514,55 @@ class ALSSuite
     assert(getFactors(model.userFactors) === getFactors(model2.userFactors))
     assert(getFactors(model.itemFactors) === getFactors(model2.itemFactors))
   }
+
+  test("StorageLevel param") {
+    // test invalid param values
+    intercept[IllegalArgumentException] {
+      new ALS().setIntermediateRDDStorageLevel("foo")
+    }
+    intercept[IllegalArgumentException] {
+      new ALS().setIntermediateRDDStorageLevel("NONE")
+    }
+    intercept[IllegalArgumentException] {
+      new ALS().setFinalRDDStorageLevel("foo")
+    }
+    // test StorageLevels
+    val sqlContext = this.sqlContext
+    import sqlContext.implicits._
+    val (ratings, _) = genExplicitTestData(numUsers = 2, numItems = 2, rank = 1)
+    val data = ratings.toDF
+    val als = new ALS().setMaxIter(1)
+    als.fit(data)
+    val factorRDD = sc.getPersistentRDDs.collect {
+      case (id, rdd) if rdd.name == "userFactors" => rdd
+    }.head
+    assert(factorRDD.getStorageLevel == StorageLevel.MEMORY_AND_DISK)
```

Comment:
* Also check `itemFactors`.
* The default storage level of intermediate RDDs are not checked.